design/ceph/submariner-globalnet.md

```diff
@@ -0,0 +1,69 @@
+
+In order to support Kubernetes clusters connected by an MCS API-compatible application, Rook needs to use "clusterset" IPs instead of the Services' cluster IPs.
+
+## Prerequisite
+
+Peer clusters should be connected using an MCS API compatible application. For example [Submariner Globalnet](https://submariner.io/getting-started/quickstart/)
+
+## Service of OSD pods
+
+For scenarios like RBD mirroring, peers need direct access to Ceph OSDs. Each OSD will have to have a standard ClusterIP Service created for it to allow this. The OSD Service will be created only when multi-cluster networking support is enabled.
+
+The reference implementation used for development of this feature is [Submariner](https://submariner.io/getting-started/quickstart/). In the reference implementation, it is important for Services to be of `type` ClusterIP. Headless Services don't have internal routing between OSDs local to a cluster or any other Ceph daemons local to the cluster.
+
+## API Changes
+
+Provide an option to enable `multiClusterService` in the `cephCluster` CR
+
+``` yaml
+spec:
+    network:
+        ### Enable multiClusterService to export the Services between peer clusters
+        multiClusterService:
+          enabled: true
+```
+
+## Service Export
+
+- `ServiceExport` CR is used to specify which services should be exposed across all the clusters in the cluster set.
+- The exported service becomes accessible as `<service>.<ns>.svc.clusterset.local`.
+- Create ServiceExport resource for each mon and OSD service.
+```yaml
+apiVersion: multicluster.x-k8s.io/v1alpha1
+kind: ServiceExport
+  name: <name>
+  namespace: rook-ceph
+```
+- Here, the ServiceExport resource name should be the name of the service that should be exported.
+
+## Implementation Details
+
+- For each mon and OSDs service:
+    - Create a corresponding `ServiceExport` resource.
+    - Verify the status of the `ServiceExport` from the `ServiceExport` resource status. Sample Status:
+    ```yaml
+    Status:
+    Conditions:
+        Last Transition Time:  2020-12-01T12:35:32Z
+        Message:               Awaiting sync of the ServiceImport to the broker
+        Reason:                AwaitingSync
+        Status:                False
+        Type:                  Valid
+        Last Transition Time:  2020-12-01T12:35:32Z
+        Message:               Service was successfully synced to the broker
+        Reason:
+        Status:                True
+        Type:                  Valid
+    ```
+    - Obtain the global IP for the exported service by issuing a DNS query to `<service>.<ns>.svc.clusterset.local`.
+    - Use this IP in the `--public-addr` flag when creating the mon or OSD deployment.
```

Comment:
How about adding this bullet as well?
- The OSD will bind to the pod IP with the new OSD flag `--public-bind-addr`.